jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java

```diff
@@ -0,0 +1,202 @@
+                        "modal-dialog-centered"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalVerticalAlignmentData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        "modal-vertical-content-1"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        "modal-vertical-content-2"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalDismissData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.dismissModal1Cross,
+                        "modal-vertical-content-1"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.dismissModal2Cross,
+                        "modal-vertical-content-2"
+                }
+        };
+    }
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        bsPage.shouldBeOpened();
+    }
+
+    @Test(dataProvider = "modalBasicData")
+    public void modalBasicFunctionalityTest(Button showButton,
+                                            Button dismissButton,
+                                            Modal modal,
+                                            String modalId) {
+        WebDriverWait wait = new WebDriverWait(WebDriverFactory.getDriver(), 5);
+
+        showButton.show();
+        showButton.click();
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id(modalId)));
+
+        modal.is().displayed();
+
+        dismissButton.show();
+        dismissButton.click();
+
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id(modalId)));
+
+        modal.is().hidden();
+    }
+
+    @Test(dataProvider = "modalCssData")
+    public void modalCssTest(Button showButton,
+                             Button dismissButton,
+                             Modal modal,
+                             String modalCss,
+                             String modalBgCss) {
+        WebDriverWait wait = new WebDriverWait(WebDriverFactory.getDriver(), 5);
+
+        showButton.show();
+        showButton.click();
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id(modalCss)));
+
+        modal.core().hasClass(modalBgCss);
+
+        dismissButton.show();
+        dismissButton.click();
+
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id(modalCss)));
```

Comment:
jdi check instead

Reply:
I have already tried it.
The driver doesn't **actually** wait whenever I use a JDI check.

Reply:
Seems I was wrong. Refactored it.